firehose.c

```diff
@@ -106,10 +111,15 @@ static int firehose_generic_parser(xmlNode *node, void *data)
 
 	if (xmlStrcmp(node->name, (xmlChar*)"log") == 0) {
 		printf("LOG: %s\n", value);
-		return 0;
+		return -EAGAIN;
 	}
 
-	return xmlStrcmp(value, (xmlChar*)"ACK") == 0 ? 1 : -1;
+	if (xmlStrcmp(value, (xmlChar*)"ACK") == 0)
+		return FIREHOSE_ACK;
+	if (xmlStrcmp(value, (xmlChar*)"NAK") == 0)
+		return FIREHOSE_NAK;
+
```

Comment:
should we print whatever came out here?

Reply:
I think we should follow up and adjust the error prints throughout the file to give better error messages. Perhaps strerror(ret) vs "device rejected" printing.